docusaurus/docs/protocol/tokenomics/token_logic_modules.md

```diff
@@ -0,0 +1,301 @@
+---
+title: Token Logic Modules
+sidebar_position: 1
+---
+
+# Token Logic Modules <!-- omit in toc -->
+
+- [Introduction](#introduction)
+- [Background: Max Claimable Amount](#background-max-claimable-amount)
+- [TLM (pre) Processing](#tlm-pre-processing)
+- [TLM: Mint=Burn (MEB)](#tlm-mintburn-meb)
+- [TLM: Global Mint (GM)](#tlm-global-mint-gm)
+- [TLM: Global Mint Reimbursement Request (GMRR)](#tlm-global-mint-reimbursement-request-gmrr)
+
+## Introduction
+
+:::warning
+
+Note that this is an active WIP and the [resources here](./resources.md) are the best starting references to learn more.
+
+In particular:
+
+- This is better than what exists in production today
+- This should be seen as an interim solution to avoid further delays
+- Implicit QoS is the best solution, but adds a lot of complexity alongside the existing upgrade
+
+:::
+
+Token Logic Module (TLM) processing consists of the following sequential steps:
+
+1. `TLM pre-processing` - General pre-processing to determine the number of tokens to settle per claim.
+2. `TLM processing` - Iterating through each TLM, sequentially.
+
+## Background: Max Claimable Amount
+
+_tl;dr Max Claimable Amount ∝ (Application Stake / Number of Suppliers per Session)_
+
+Per **Algorithm 1** of the [Relay Mining paper](https://arxiv.org/pdf/2305.10672),
+the maximum amount a supplier can claim from an application in a single session is
+proportional to the Application's stake divided by the number of suppliers in the session.
+
+This is referred to as "Relay Mining Payable Relay Accumulation" in the paper and
+is described by the following pseudo-code:
+
+![Algorithm 1](https://github.com/user-attachments/assets/d1a61535-aa31-447d-88ea-c8d14dcb20c6)
+
+## TLM (pre) Processing
+
+_tl;dr Determine if the claim settlement amount is greater than the maximum claimable amount prior to running each individual TLM._
+
+**Prior to** processing each individual TLM, we need to understand if the amount claimed
+by the supplier adheres to the optimistic maxima set per the limits of the Relay Mining algorithm.
+
+Suppliers always have the option to over-service an application (**i.e. do free work**),
+in order to ensure high quality service in the network. This may lead to offchain
+reputation benefits (e.g. Gateways favoring them), but suppliers' on-chain rewards
+are always limited by the cumulative amounts applications' stakes (at session start; per service)
```

Comment:
> per service

To have a `Maximum Claimable Amount` that mitigates the `Application`s stake to go negative; its stake needs to be **per service**

Otherwise `Application`s will still be able to over-service across different `Services` and getting more work done (beyond the _Relay Accuracy_ parameter) than what their stake permits.

If `Maximum Claimable Amount` does not enforce this, I don't see how it could be useful.

Reply:
Great callout!